x.ps1

```diff
@@ -24,5 +24,12 @@ foreach ($python in "py", "python3", "python", "python2") {
     }
 }
 
+$found = (Get-Command 'python*' -CommandType Application -ErrorAction SilentlyContinue | Where-Object {$_.name -match 'python\d'})
```

Comment:
> The Microsoft Store version of Python installs itself as pythonM.m, with M being the major version and m the minor.

I don't think this code is doing what you expect. It only matches `python[0-9]`, which doesn't include the trailing `.m`, right? or does `Where-Object` do a substring match?

Reply:
The `-match` operator does a regex match. Since no `^` was put at the start nor `$` at the end, this ends up as a substring match, matching any string that contains the word `python` immediately followed by a digit.

Reply:
Ok, thanks. Let's change this to require it to start with `python`, then.
```suggestion
$found = (Get-Command 'python*' -CommandType Application -ErrorAction SilentlyContinue | Where-Object {$_.name -match '^python\d'})
```